sdk/communication/azure-communication-sms/azure/communication/sms/_shared/models.py

```diff
@@ -173,19 +199,81 @@ def __init__(self, user_id, **kwargs):
         if self.raw_id is None:
             self.raw_id = _microsoft_teams_user_raw_id(self)
 
+    def __eq__(self, other):
+        return self.raw_id == _microsoft_teams_user_raw_id(other)
+
 
 def _microsoft_teams_user_raw_id(identifier: MicrosoftTeamsUserIdentifier) -> str:
     user_id = identifier.properties['user_id']
     if identifier.properties['is_anonymous']:
-        return '8:teamsvisitor:{}'.format(user_id)
+        return f'{TEAMS_USER_ANONYMOUS_PREFIX}{user_id}'
     cloud = identifier.properties['cloud']
     if cloud == CommunicationCloudEnvironment.DOD:
-        return '8:dod:{}'.format(user_id)
+        return f'{TEAMS_USER_DOD_CLOUD_PREFIX}{user_id}'
     elif cloud == CommunicationCloudEnvironment.GCCH:
-        return '8:gcch:{}'.format(user_id)
+        return f'{TEAMS_USER_GCCH_CLOUD_PREFIX}{user_id}'
     elif cloud == CommunicationCloudEnvironment.PUBLIC:
-        return '8:orgid:{}'.format(user_id)
-    return '8:orgid:{}'.format(user_id)
+        return f'{TEAMS_USER_PUBLIC_CLOUD_PREFIX}{user_id}'
+    return f'{TEAMS_USER_PUBLIC_CLOUD_PREFIX}{user_id}'
+
+
+MicrosoftBotProperties = TypedDict(
+    'MicrosoftBotProperties',
+    bot_id=str,
+    is_global=bool,
+    cloud=Union[CommunicationCloudEnvironment, str]
+)
+
+
+class MicrosoftBotIdentifier(object):
+    """Represents an identifier for a Microsoft bot.
+
+    :ivar str raw_id: Optional raw ID of the identifier.
+    :ivar kind: The type of identifier.
+    :vartype kind: str or CommunicationIdentifierKind
+    :ivar Mapping properties: The properties of the identifier.
```

Comment:
pls update the type to MicrosoftBotProperties as Anna mentioned it during the review

Reply:
```suggestion
    :ivar MicrosoftBotProperties properties: The properties of the identifier.
```
fyi 
![image](https://user-images.githubusercontent.com/26452616/224945878-4700900f-b298-4050-bcf7-3564b0cb2231.png)

Reply:
Updated